client/commands/v2/pysa_server.py

```diff
@@ -5,80 +5,71 @@
 
 
 """
-This is a temporary implementation of Pysa's language server. This is an exact copy
-of Pyre's language server, which lives in persistent.py.
+This is an implementation of Pysa's language server. It is a refactored
+version of persistent.py.
 """
 
 import asyncio
 import logging
-import os
-import traceback
-from pathlib import Path
-from typing import Optional, AsyncIterator, Sequence, Dict
 
 from ... import (
     json_rpc,
-    error,
     command_arguments,
     commands,
     configuration as configuration_module,
 )
 from . import (
     language_server_protocol as lsp,
-    server_connection,
     async_server_connection as connection,
     start,
-    stop,
-    incremental,
 )
 from .persistent import (
     LSPEvent,
     ServerState,
-    CONSECUTIVE_START_ATTEMPT_THRESHOLD,
     _read_lsp_request,
     try_initialize,
-    _start_pyre_server,
     _log_lsp_event,
-    parse_subscription_response,
-    StartSuccess,
-    StartFailure,
     _publish_diagnostics,
-    type_errors_to_diagnostics,
     InitializationExit,
     InitializationSuccess,
     InitializationFailure,
 )
 
-
 LOG: logging.Logger = logging.getLogger(__name__)
 
 
-class PysaServerHandler(connection.BackgroundTask):
-    binary_location: str
-    server_identifier: str
-    pyre_arguments: start.Arguments
-    client_output_channel: connection.TextWriter
-    server_state: ServerState
+class PysaServer:
+    # I/O Channels
+    input_channel: connection.TextReader
+    output_channel: connection.TextWriter
+
+    # Immutable States
+    client_capabilities: lsp.ClientCapabilities
+
+    # Mutable States
+    state: ServerState
```

Comment:
It looks like our linter is not happy because this is never used. Could you delete it?